File: src/main/java/upbrella/be/user/dto/response/SessionUser.java
package upbrella.be.user.dto.response;

import lombok.Builder;
import lombok.Getter;

import java.io.Serializable;
import upbrella.be.user.entity.User;

@Getter
@Builder
public class SessionUser implements Serializable {

    private long id;
    private long socialId;
    private Boolean adminStatus;

    public static SessionUser fromUser(User user) {

        return SessionUser.builder()
                .id(user.getId())
                .adminStatus(user.getAdminStatus())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/user/dto/response/SessionUser.kt
package upbrella.be.user.dto.response

import upbrella.be.user.entity.User
import java.io.Serializable

data class SessionUser(
    val id: Long,
    val socialId: Long? = null,
    val adminStatus: Boolean?
) : Serializable {
    companion object {
        fun fromUser(user: User): SessionUser {
            return SessionUser(
                id = user.id!!,
                socialId = user.socialId,
                adminStatus = user.adminStatus
            )
        }
    }
}
